modules/n1ql/pages/n1ql-language-reference/objectfun.adoc

```diff
@@ -1355,6 +1460,99 @@ LIMIT 1;
 ----
 ====
 
+[[fn-obj-type,OBJECT_TYPES()]]
+== OBJECT_TYPES(`expression`)
+
+=== Description
+
+This function takes an object and returns a new one where all the values are replaced with their data types as reported by the TYPE() function.
+
+=== Arguments
+
+expression:: An expression representing an object.
+
+=== Return Value
+
+An object with the same fields as the input object, but with all values replaced by their corresponding data types. 
+
+=== Example
+
+[[obj-type-ex,OBJECT_TYPES() Example]]
+====
+.Query
+[source,sqlpp]
+----
+SELECT OBJECT_TYPES({"flight":"AI444","duration":180,"gate":NULL,"aircraft":{"model":"Boeing 737"}}) 
+    AS object_types;
+----
+
+.Results
+[source,json]
+----
+[
+    {
+        "object_types":{
+            "flight":"string",
+            "duration":"number",
+            "gate":"null",
+            "aircraft":"object"
+        }
+    }
+]
+----
+====
+
+[[fn-obj-types-nested,OBJECT_TYPES_NESTED()]]
+== OBJECT_TYPES_NESTED(`expression`)
+
+=== Description
+
+This function takes an object and returns a new one where all non-composite field values replaced with their data types as reported by the TYPE() function. 
+Additionally, all composite type fields (ARRAYs and OBJECTs) are recursively processed in the same way, ensuring that all nested values are also replaced with their types. 
```

Comment:
Tweaking the wording slightly so that the description (hopefully) explains why you would use the function.  Added link to TYPES() function.

```suggestion
This function returns the data type of every non-composite field in the supplied object, as reported by the xref:n1ql-language-reference/typefun.adoc#fn-type-type[TYPE()] function.
Additionally, all composite type fields (ARRAYs and OBJECTs) are recursively processed in the same way, returning the data types of all nested values. 
```